src/app/FakeLib/Vb6helper.fs

```diff
@@ -0,0 +1,307 @@
+///
+/// Used to temporarily register any .net dependencies before running 
+/// a VB6 build
+let public RegisterAssembliesWithCodebase workingDir (assemblies:string seq) =
+    let registerAssemblyWithCodebase assembly =
+        async {
+            let! regAsmResult = 
+                asyncShellExec {defaultParams with 
+                                    Program = regAsmToolPath
+                                    WorkingDirectory = workingDir
+                                    CommandLine = (sprintf "\"%s\" /tlb:%s /codebase" assembly ((Path.GetFileName assembly) + ".tlb"))
+                                }
+            if regAsmResult <> 0 then failwith (sprintf "Register %s with codebase failed" assembly)
+            return ()
+        }
+    assemblies
+    |> Seq.map registerAssemblyWithCodebase
+    |> Async.Parallel
+    |> Async.RunSynchronously
+    |> ignore
+
+/// Executes `Regasm.exe` with the `/codebase /tlb /unregister` options
+///
+/// Used to unregegister any temporarily registerd .net dependencies
+/// _after_ running a VB6 build
+let public UnregisterAssemblies workingDir (assemblies:string seq) =
+    let registerAssemblyWithCodebase assembly =
+        async {
+            let! regAsmResult = 
+                asyncShellExec {defaultParams with 
+                                    Program = regAsmToolPath
+                                    WorkingDirectory = workingDir
+                                    CommandLine = (sprintf "\"%s\" /tlb:%s /codebase /unregister"  assembly ((Path.GetFileName assembly) + ".tlb"))
+                                }
+            if regAsmResult <> 0 then failwith (sprintf "Unregister %s with codebase failed" assembly)
+            return ()
+        }
+    assemblies
+    |> Seq.map registerAssemblyWithCodebase
+    |> Async.Parallel
+    |> Async.RunSynchronously
+    |> ignore
+
+/// All-In-one build and manifest function for VB6 __applications__ referencing .net __libraries__
+///
+/// ## Paramteters
+///  - `getConfig`- function to alter default VB6 build parameters
+///  - `vb6Projects` - Paths to all `.vbp` files to build
+///  - `possibleAssemblies` - Paths to assemblies that may be referenced by the VB6 projects
+///
+/// ## Process
+///
+/// This function will:
+///
+/// 1. Determine which of the `possibleAssemnblies` are referenced by any of the provided `.vbp` files
+/// 2. Temporarily register any referenced assemblies using `RegAsm /codebase /tlb`
+/// 3. Run VB6 command line make on all provided `.vbp` projects
+/// 4. Unregister all registered assemblies
+/// 5. Generate and embed Side-By-Side interop appplication manifests in all generated VB6 executables
+/// 6. Generate and embed Side-By-Side interop assembly manifest in all referenced assemblies 
+let public BuildAndEmbedInteropManifests (getConfig: Vb6BuildParams->Vb6BuildParams) (vb6Projects: string seq) (possibleAssemblies: string seq) =
+    let config = defaultVb6BuildParams |> getConfig 
```

Comment:
For tasks please add trace start/end functions like in

https://github.com/fsharp/FAKE/blob/master/src/app/FakeLib/FixieHelper.fs#L38
https://github.com/fsharp/FAKE/blob/master/src/app/FakeLib/FixieHelper.fs#L59